--- halide-2019.08.27.orig/CMakeLists.txt
+++ halide-2019.08.27/CMakeLists.txt
@@ -80,16 +80,22 @@ file(TO_NATIVE_PATH "${LLVM_TOOLS_BINARY
 file(TO_NATIVE_PATH "${LLVM_TOOLS_BINARY_DIR}/clang${CMAKE_EXECUTABLE_SUFFIX}" CLANG)
 file(TO_NATIVE_PATH "${LLVM_TOOLS_BINARY_DIR}/llvm-config${CMAKE_EXECUTABLE_SUFFIX}" LLVM_CONFIG)
 
-# LLVM doesn't appear to expose --system-libs via its CMake interface,
-# so we must shell out to llvm-config to find this info
-execute_process(COMMAND ${LLVM_CONFIG} --system-libs --link-static OUTPUT_VARIABLE HALIDE_SYSTEM_LIBS_RAW)
-string(STRIP "${HALIDE_SYSTEM_LIBS_RAW}" HALIDE_SYSTEM_LIBS_RAW)  # strip whitespace from start & end
-string(REPLACE " " ";" HALIDE_SYSTEM_LIBS "${HALIDE_SYSTEM_LIBS_RAW}")  # convert into a list
-if("${HALIDE_SYSTEM_LIBS}" STREQUAL "")
-  # It's theoretically possible that this could be legitimately empty,
-  # but in practice that doesn't really happen, so we'll assume it means we
-  # aren't configured correctly.
-  message(WARNING "'llvm-config --system-libs --link-static' is empty; this is possibly wrong.")
+if(HALIDE_SHARED_LLVM_LIBRARY)
+  # Since we will be linking to the shared LLVM library,
+  # we will get all the transitive dependencies for free automagically.
+  set(HALIDE_SYSTEM_LIBS)
+else()
+  # LLVM doesn't appear to expose --system-libs via its CMake interface,
+  # so we must shell out to llvm-config to find this info
+  execute_process(COMMAND ${LLVM_CONFIG} --system-libs --link-static OUTPUT_VARIABLE HALIDE_SYSTEM_LIBS_RAW)
+  string(STRIP "${HALIDE_SYSTEM_LIBS_RAW}" HALIDE_SYSTEM_LIBS_RAW)  # strip whitespace from start & end
+  string(REPLACE " " ";" HALIDE_SYSTEM_LIBS "${HALIDE_SYSTEM_LIBS_RAW}")  # convert into a list
+  if("${HALIDE_SYSTEM_LIBS}" STREQUAL "")
+    # It's theoretically possible that this could be legitimately empty,
+    # but in practice that doesn't really happen, so we'll assume it means we
+    # aren't configured correctly.
+    message(WARNING "'llvm-config --system-libs --link-static' is empty; this is possibly wrong.")
+  endif()
 endif()
 
 # Check LLVM
@@ -175,6 +181,7 @@ option(TARGET_OPENGL "Include OpenGL/GLS
 option(TARGET_OPENGLCOMPUTE "Include OpenGLCompute target" ON)
 option(TARGET_D3D12COMPUTE "Include Direct3D 12 Compute target" ON)
 option(HALIDE_SHARED_LIBRARY "Build as a shared library" ON)
+option(HALIDE_SHARED_LLVM_LIBRARY "Use shared versions of LLVM libraries" OFF)
 option(HALIDE_ENABLE_RTTI "Enable RTTI" ${LLVM_ENABLE_RTTI})
 option(HALIDE_ENABLE_EXCEPTIONS "Enable exceptions" ${LLVM_ENABLE_EH})
 option(HALIDE_USE_CODEMODEL_LARGE "Use the Large LLVM codemodel" OFF)
--- halide-2019.08.27.orig/src/CMakeLists.txt
+++ halide-2019.08.27/src/CMakeLists.txt
@@ -759,25 +759,10 @@ else()
   endif()
 endif()
 
-# Get the LLVM libraries we need
-llvm_map_components_to_libnames(LIBS ${LLVM_COMPONENTS})
-
-# When building a shared library the LLVM libraries will be
-# embedded in the Halide library. When building a static library
-# LLVM is not embedded but CMake knows that when building an executable
-# against the Halide static library that it needs to link LLVM too so
-# PRIVATE scope is the correct choice here.
-target_link_libraries(Halide PRIVATE ${LIBS})
-
-if (NOT MSVC)
-  set(LLVM_CONFIG ${LLVM_TOOLS_BINARY_DIR}/llvm-config)
-  execute_process(COMMAND "${LLVM_CONFIG}" --system-libs ${LLVM_COMPONENTS} OUTPUT_VARIABLE EXTRA_LIBS)
-  string(STRIP EXTRA_LIBS "${EXTRA_LIBS}")
-  string(REPLACE "-l" ";" EXTRA_LIBS "${EXTRA_LIBS}")
-  string(REPLACE "\n" "" EXTRA_LIBS "${EXTRA_LIBS}")
-  string(REPLACE " " "" EXTRA_LIBS "${EXTRA_LIBS}")
-  target_link_libraries(Halide PUBLIC ${EXTRA_LIBS})
+if(HALIDE_SHARED_LLVM_LIBRARY)
+  set(LLVM_USE_SHARED "USE_SHARED")
 endif()
+llvm_config(Halide ${LLVM_USE_SHARED} ${LLVM_COMPONENTS})
 
 install(TARGETS Halide
         RUNTIME DESTINATION bin
